### PR TITLE
disable "ssl root certificate" settings if user can't mount external storages

### PR DESCRIPTION
**Pull request by schiessle**
only add the possibility to import ssl root certificates at the personal setting if the user can mount external storages

cc @MTRichards @LukasReschke @CaptionF

### CODE REVIEW

**Comment by LukasReschke** [comment]
>  I only checked the menu. I will update the PR

:-) - it's quite some confusing black magic what we do with the menu there, yes :speak_no_evil:

File: settings/personal.php [hunk 1]
```diff
@@ -156,8 +156,17 @@
 
 $formsAndMore = array_merge($formsAndMore, $formsMap);
 
-// add bottom hardcoded forms from the template
-$formsAndMore[]= array( 'anchor' => 'ssl-root-certificates', 'section-name' => $l->t('SSL root certificates') );
+// add bottom hardcoded forms from the template if external storages are enabled
+// for the user
+$externalStorageEnabled = \OC::$server->getAppManager()->isEnabledForUser('files_external');
```
Comment (LukasReschke) [inline settings/personal.php]: I'm not sure if `isEnabledForUser` is the right action here. `files_external` works on a global base and can only be enabled for all users.

I think what we want here is to check if  `\OC_Mount_Config::getPersonalBackends` is an empty array

Comment (LukasReschke) [inline settings/personal.php]: See also https://github.com/owncloud/core/blob/728780aee8dbbdb41746b7510096c8690505a991/apps/files_external/lib/config.php#L325-325

Comment (LukasReschke) [inline settings/personal.php]: Ah. I see, we already do that below. This can stay then. :speak_no_evil:

**Comment by schiesbn** [comment]
Now also the actual setting should be gone

**Comment by LukasReschke** [comment]
Can we have this in `\OC\Settings\Controller\CertificateController` as well? If the frontend is blocked then we should also block the backend. A simple `return new JSONResponse('You shall not pass');` is fine ;)

**Comment by schiesbn** [comment]
@LukasReschke the last commit is just for you... ;)

**Comment by LukasReschke** [comment]
I shall approve this one then :+1:

**Comment by schiesbn** [comment]
If Jenkins finish successfully I will prepare a backport for 8.1.2 cc @karlitschek

**Comment by schiesbn** [comment]
@owncloud-bot retest this please

**Comment by schiesbn** [comment]
ok, backport for 8.1.2 it here: https://github.com/owncloud/core/pull/18064

**Comment by RealRancor** [comment]
Maybe to late for the party and i'm not also familiar with the code but what about federated clouds? Isn't this also needed for federation and not only for external storages?

**Comment by codeling** [comment]
@RealRancor @schiesbn yes just had this same problem - federated cloud requires a root certificate (e.g. cacert) to be added, but you have to enable external storage app to be able to add custom certificates!

**Comment by RealRancor** [comment]
@codeling Probably the best to create a new issue for this.

**Comment by schiesbn** [comment]
@codeling @RealRancor good point, please open a new issue for it. Thanks!

**Pull request merged**